File: BotCommands-commands/text/src/javaDocExamples/java/doc/java/examples/filters/MyTextCommandFilter.java
package doc.java.examples.filters;

import io.github.freya022.botcommands.api.commands.text.TextCommandFilter;
import io.github.freya022.botcommands.api.commands.text.TextCommandVariation;
import io.github.freya022.botcommands.api.core.service.annotations.BService;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

@BService
@NullMarked
public class MyTextCommandFilter implements TextCommandFilter {

    @Override
    public boolean getGlobal() {
        return true;
    }

    @Nullable
    @Override
    public String check(MessageReceivedEvent event, TextCommandVariation commandVariation, String args) {
        if (event.getChannel().getIdLong() != 722891685755093076L) {
            event.getMessage().reply("Can only run commands in <#722891685755093076>").queue();
            return "Wrong channel";
        }
        return null;
    }
}

File: BotCommands-core/src/kotlinDocExamples/kotlin/doc/kotlin/examples/filters/MyTextCommandFilter.kt
package doc.kotlin.examples.filters

import dev.freya02.botcommands.jda.ktx.coroutines.await
import io.github.freya022.botcommands.api.commands.text.TextCommandFilter
import io.github.freya022.botcommands.api.commands.text.TextCommandVariation
import io.github.freya022.botcommands.api.core.service.annotations.BService
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

@BService
class MyTextCommandFilter : TextCommandFilter {

    override val global: Boolean get() = true

    override suspend fun checkSuspend(
        event: MessageReceivedEvent,
        commandVariation: TextCommandVariation,
        args: String
    ): String? {
        if (event.channel.idLong != 722891685755093076) {
            event.message.reply("Can only run commands in <#722891685755093076>").await()
            return "Wrong channel"
        }
        return null
    }
}
